src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1261,63 +1268,86 @@ private AuthenticationResult localFlow(CallbackHandler callbackHandle,
-                request.setRequestId(callbackHandle.callback.hashCode());
-                Logger.v(TAG, "Starting Authentication Activity with callback:"
-                        + callbackHandle.callback.hashCode());
-                putWaitingRequest(callbackHandle.callback.hashCode(),
-                        new AuthenticationRequestState(callbackHandle.callback.hashCode(), request,
-                                callbackHandle.callback));
-
-                if (useDialog) {
-                    AuthenticationDialog dialog = new AuthenticationDialog(mHandler, mContext,
-                            this, request);
-                    dialog.show();
-                } else {
-                    // onActivityResult will receive the response
-                    if (!startAuthenticationActivity(activity, request)) {
-                        callbackHandle.onError(new AuthenticationException(
-                                ADALError.DEVELOPER_ACTIVITY_IS_NOT_RESOLVED));
-                    }
-                }
+                acquireTokenInteractively(callbackHandle, activity, request, useDialog);
             } else {
-
+                final String errorInfo = authResult == null? "" : authResult.getErrorLogInfo();
                 // User does not want to launch activity
-                Logger.e(TAG, "Prompt is not allowed and failed to get token:", "",
+                Logger.e(TAG, "Prompt is not allowed and failed to get token:", request.getLogInfo() + " " + errorInfo,
                         ADALError.AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED);
                 callbackHandle.onError(new AuthenticationException(
-                        ADALError.AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED));
+                        ADALError.AUTH_REFRESH_FAILED_PROMPT_NOT_ALLOWED, request.getLogInfo() + " " + errorInfo));
             }
         }
 
         return null;
     }
 
+    private void acquireTokenInteractively(final CallbackHandler callbackHandle, final IWindowComponent activity, 
+            final AuthenticationRequest request, final boolean useDialog) {
+        //Check if there is network connection
+        if (!mConnectionService.isConnectionAvailable()) {
+            AuthenticationException exc = new AuthenticationException(
+                    ADALError.DEVICE_CONNECTION_IS_NOT_AVAILABLE,
+                    "Connection is not available to request token");
+            Logger.w(TAG, "Connection is not available to request token", request.getLogInfo(),
+                    ADALError.DEVICE_CONNECTION_IS_NOT_AVAILABLE);
+            callbackHandle.onError(exc);
+            return;
+        }
+        
+        // start activity if other options are not available
+        // delegate map is used to remember callback if another
+        // instance of authenticationContext is created for config
+        // change or similar at client app.
+        mAuthorizationCallback = callbackHandle.callback;
```

Comment:
I don't think there is any need to store the callback as a member. But I don't want to change the behavior for the current code unless we get time looking into it.